### PR TITLE
Fixes memory leaks and uninitialized variable

### PR DESCRIPTION
**Pull request by rahul003**
## Description ##
Fixing errors identified by static analysis tool Coverity.

- Fixes memory leaks
- Sets value to uninitialized variable
- Adds a default case to a switch statement

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] For user-facing API changes, API doc string has been updated. For new C++ functions in header files, their functionalities and arguments are well-documented. 
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Fixes memory leaks
- [x] Sets value to uninitialized variable
- [x] Adds a default case to a switch statement

### CODE REVIEW

**Comment by rahul003** [comment]
Can someone help me understand why VC++ says more than one default for switch case?

**Comment by rahul003** [comment]
Can someone help me understand why VC++ says more than one default for switch case? There is only one in stream_manager.h

File: src/engine/stream_manager.h [hunk 1]
```diff
@@ -117,6 +120,9 @@ RunContext StreamManager<kNumGpus, kStreams>::GetIORunContext(
       LOG(FATAL) << MXNET_GPU_NOT_ENABLED_ERROR;
```
Comment (reminisce) [inline src/engine/stream_manager.h]: Try adding a `break` here and see what VC++ says?

Comment (rahul003) [reply src/engine/stream_manager.h]: Thanks, but this doesn't result in any change. It still complains about multiple default cases

**Pull request closed without merging**